.github/workflows/typechecks.yml

```diff
@@ -0,0 +1,43 @@
+name: Tests, Linter & Typecheck
+
+on: pull_request
+
+jobs:  
+  lint:
+    name: 'Linting Markdown'
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout Repo
+        uses: actions/checkout@v2
+
+      - name: Use Node.js 16.7
+        uses: actions/setup-node@v2
+        with:
+          node-version: 16.7
+
+      - run: npm install -g markdownlint-cli@0.23.2
+      - run: markdownlint '**/*.md' --ignore node_modules
+  test:
```

Comment:
why is this called test if it only does typechecking?

Reply:
No reason I can think of, can change it to `typecheck`

Reply:
Well it is "testing" whether the types are valid... 😄

Reply:
yeah I can buy that. maybe this should stay as `test`.